src/cart/crt_swim.c

```diff
@@ -738,8 +736,6 @@ int crt_swim_rank_add(struct crt_grp_priv *grp_priv, d_rank_t rank)
 		D_FREE(cst);
 
 	if (rc && rc != -DER_ALREADY) {
-		if (rank_in_list)
-			crt_swim_rank_del(grp_priv, rank);
```

Comment:
this change maybe better to confirm with @dmiter

Reply:
This is the defect type of dead code: rank_in_list is set to TRUE only when rc==0, so when the code path gets to here in out: with rc!=0, then rank_in_list must still be false as initialized.

Reply:
ok